biz.aQute.bndlib.tests/src/test/ProjectTest.java

```diff
@@ -762,6 +765,26 @@ public void testMultidirsrc() throws Exception {
 			.isFile());
 	}
 
+	/*
+	 * Verify that this also works when you have multiple directories
+	 */
+	public void testSrcPointsToFile() throws Exception {
+		Workspace ws = getWorkspace("testresources/ws");
+		Project p = ws.getProject("pfilesrc");
+		Collection<File> sourcePath = p.getSourcePath();
+		File a = p.getFile("a");
+		File b = p.getFile("b");
+		String expectedError = "the src path (src property) contains an entry that is not a directory "
```

Comment:
Generally our tests use the `check` method to test for the state of the processor and ensure certain messages are present. You will find examples of the use of the check method in many other tests.

Reply:
Happy to use `check()` if that's what you decide, but I offer the following observations of pros & cons vs the AssertJ approach that I used:

1. `check()` and `containsExactly()` will both give essentially the same diagnostic information about why the check failed (ie, which patterns were missing, and which elements were found that shouldn't have been). However, `containsExactly()` will report the diagnostic in the actual `AssertionError` instead of to stderr. Thus it won't pollute stderr with output that is subsequently hard to find (especially when running tests in bulk). When using it integrated with Eclipse JUnit runner, it also means that if the assertion fails, you'll be able to click on the failed test and immediately see the corresponding diagnostic message instead of having to wade through stderr to find it.
2. `check()` doesn't check to see if the error(s)' location(s) have been set properly, like this implementation does.
3. `check()` doesn't facilitate checking the severity level of the individual problems - it matches patterns against both errors and warnings. As the test currently stands, the test is a guarantee that the problem was reported as an error; if I use `check()` instead then the test can't guarantee that the problem wasn't reported as a warning instead of an error.

Anyway, I am new on this project and you're the boss - if you feel strongly about using `check()`, or if there are other reasons for using `check()` that I am not aware of, I'm more than happy to do it that way instead.

Reply:
I know your check is extremely precise. But precision is non-free. Do we really need to the test here to be that precise? Since the test is very focused, we only really need to see that there is an error. The precises wording of the message may get improved later and then this precise test case will break. Some lubrication is nice to reduce friction between the code and its tests. check with its pattern matching allows for some lubrication. Sometimes you need to be very precise but not always.

Reply:
I have relaxed it & used check. Would be nice if we could find an error using an RE and then have that method return the Location object associated with the matched error, so that we could check the right location is set (best of both worlds).